### PR TITLE
first version js-webapp-async

### PR DESCRIPTION
**Pull request by cwmeijer**
Wrote few intro lines and outro lines and pasted existing text from cpp2wasm in between

### CODE REVIEW

File: js-webapp-async/README.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+Notice that we also added a slider input to the page. This slider isn't connected to anything, and is there to illustrate UI blocking. Notice that while the webassembly code is still running, the slider is completely unresponsive. If this were an actual webapp and not just a demo, the UI blocking would surely annoy any users of the system and possibly make working with the app cumbersome and impracticle. (TODO: animated gif of slider not responding) We can easily solve this, and keep the UI responsive at all times, using web workers.
+
+### Web workers
+
+A web worker is an object that handles execution of a piece of code aynchronously. On completion the web worker will notify so the result can be processed and, in this case, rendered on the page.
```
Comment (sverhoeven) [inline js-webapp-async/README.md]: Please link to more info on web workers
```suggestion
A [web worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API) is an object that handles execution of a piece of code aynchronously. On completion the web worker will notify so the result can be processed and, in this case, rendered on the page. 

```

File: js-webapp-async/README.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      // this JavaScript snippet is appended to <<worker-consumer>>
+      worker.postMessage({
+        type: 'CALCULATE',
+        payload: { epsilon: 0.001, guess: -20 }
+      });
+      // this JavaScript snippet is appended to <<worker-consumer>>
+      worker.onmessage = function(message) {
+        if (message.data.type === 'RESULT') {
+          const root = message.data.payload.root;
+          document.getElementById('answer').innerHTML = root.toFixed(2);
```
Comment (sverhoeven) [inline js-webapp-async/README.md]: Use same output as blocking example page.
```suggestion
          document.getElementById("answer")
               .innerHTML = "Function root is approximately at x = " +
                            root.toFixed(2);

```

**Review by sverhoeven — APPROVED** [review]
Very good first draft. I have some small suggestions though.

I would like to move the main focus of the blog from asynchronous to background thread.
JS asynchronously to me is the use of callbacks, promises and await/async, see https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous . We use a callback for onmessage, but what is special here is that the computation is done in another thread next to the main UI thread.

**Comment by cwmeijer** [comment]
Thanks for the suggestions, Stefan. All good points of course :-) I'll continue solving the remaining TODOs, and change the text a bit towards threads, and away from async.

**Pull request merged**